File: 1/1.5/Go/1.5.go
package main

import (
	"fmt"
	"strings"
)

func areOneStepAway(s1 string, s2 string) bool {
	if s1 == s2 {
		return true
	}
	if len(s1) == len(s2) {
		count := 0
		for i, c := range []rune(s1) {
			if c != ([]rune(s2))[i] {
				count = count + 1
			}
		}
		if count == 1 {
			return true
		} else {
			return false
		}
	} else if len(s1)-len(s2) == 1 {
		for _, c := range []rune(s2) {
			if !strings.Contains(s1, string(c)) {
				return false
			}
		}
		return true
	} else if len(s2)-len(s1) == 1 {
		for _, c := range []rune(s1) {
			if !strings.Contains(s2, string(c)) {
				return false
			}
		}
		return true
	}
	return false
}

func main() {
	fmt.Println(areOneStepAway("hello", "hello"))
	fmt.Println(areOneStepAway("hello", "hell"))
	fmt.Println(areOneStepAway("hello", "help"))
	fmt.Println(areOneStepAway("hellu", "hello"))
	fmt.Println(areOneStepAway("helloo", "hello"))
	fmt.Println(areOneStepAway("hellooo", "hello"))
}

File: 1/1.5/Python/1.5.py
def areOneStepAway(s1: str, s2: str) -> bool:
    if s1 == s2:
        return True
    if len(s1) == len(s2):
        count = 0
        for i,c in enumerate(s1):
            if c != s2[i]:
                count = count + 1
        if count == 1:
            return True
        else:
            return False
    elif len(s1) - len(s2) == 1 and all(x in s1 for x in s2 ):
        return True
    elif len(s2) - len(s1) == 1 and all(x in s2 for x in s1 ):
        return True
    else:
        return False


if __name__ == '__main__':
    print(areOneStepAway('hello','hello'))
    print(areOneStepAway('hello','hell'))
    print(areOneStepAway('hello','help'))
    print(areOneStepAway('hellu','hello'))
    print(areOneStepAway('helloo','hello'))
    print(areOneStepAway('hellooo','hello'))